ssh_key_rotator/publisher/key_publisher.py

```diff
@@ -65,21 +65,39 @@ def publish_new_key(self) -> str:
         return public_key.decode()
 
 
+def create_argument_parser() -> ArgumentParser:
+    argument_parser = ArgumentParser(
+        prog="key_publisher",
+        description="Creates public/private SSH keys and publishes the public key either locally or to S3 (default is S3)",
+        epilog="Thanks for using key_publisher! :)",
+    )
+
+    parser.add_argument("user")
+    parser.add_argument("hostname")
+    parser.add_argument("-ds", "--datastore", choices=["s3", "local"], default="s3")
+
+    options = parser.add_argument_group("Options")
+    options.add_argument(
+        "-ds (s3 | local)",
+        "--datastore (s3 | local)",
+        help="choose where to store the public key, on S3 or on the local system (default is S3)"
+    )
+
+    arguments = parser.add_argument_group("Required Arguments")
+    arguments.add_argument("user")
+    arguments.add_argument("hostname")
+
+    return argument_parser
+
+
 if __name__ == "__main__":
```

Comment:
this should be rollout into a function and since this won't be a module ever - this line is not needed

ultimately you should just invoke a "main()" function

another thing to do is to have `main` include a try/except for any errors, so if things go wrong, you have a last line of defense to produce a user-friendly error message

Reply:
for dev command line purposes before production I added a publisher/__main__.py file such that we run it via ssh_key_rotator.publisher ..args